gdal/frmts/esric/esric_dataset.cpp

```diff
@@ -0,0 +1,475 @@
+//-------------------------------------------------------------------------------
+// Name: esric_dataset.cpp
+// Purpose : gdal driver for reading Esri compact cache as raseter
+//           based on public documentation available at
+//           https://github.com/Esri/raster-tiles-compactcache
+//
+// Author : Lucian Plesea
+//
+// Udate : 06 / 10 / 2020
+//
+//  Copyright 2020 Esri
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissionsand
+//  limitations under the License. ?
+//
+//-------------------------------------------------------------------------------
+
+#include <gdal_priv.h>
+#include <vector>
+#include <algorithm>
+
+using namespace std;
+
+CPL_C_START
+void CPL_DLL GDALRegisterMe();
+CPL_C_END
+
+void CPL_DLL GDALRegister_ESRIC();
```

Comment:
should be surrounded by CPL_C_START / CPL_C_END if you want to keep it for plugin declaration (similarly as done in gcore/gdal_frmts.h)